### PR TITLE
allow access to global command history when using a specifc history

### PR DESCRIPTION
**Pull request by ailin-nemui**
also add history loading method

### CODE REVIEW

**Comment by ailin-nemui** [comment]
usage: ctrl+up/down

**Comment by ailin-nemui** [comment]
sample script for your autorun:

```perl
use strict;
use warnings;
use YAML::Tiny;

my $histfile = Irssi::get_irssi_dir()."/history";

my ($hist) = eval { YAML::Tiny::LoadFile($histfile) };
if ($hist) {
    Irssi::UI::Window::load_history_entries(undef, @$hist);
}

Irssi::signal_add 'gui exit' => sub {
    if (Irssi::settings_get_bool('settings_autosave')) {
	YAML::Tiny::DumpFile($histfile, [ Irssi::UI::Window::get_history_entries(undef) ] )
    }
}
```

File: src/fe-common/core/command-history.c [hunk 1]
```diff
@@ -30,32 +30,108 @@
...
+static void history_entry_destroy(HISTORY_ENTRY_REC *entry)
+{
+	g_free((char *)entry->text);
+	g_free(entry);
+}
+
+#define history_list_last command_history_list_last
+#define history_list_first command_history_list_first
+#define history_list_prev command_history_list_prev
+#define history_list_next command_history_list_next
```
Comment (dequis) [inline src/fe-common/core/command-history.c]: Huh. Why?

**Comment by dequis** [comment]
Good stuff.

The commit log / PR title could use some more love. I mean, global window history is the default behavior, but being able to use global history while window history is enabled is a very good feature.

Not sure I fully understand what g_history is supposed to hold.

Leaving the job of persisting history to a perl script is a weird choice (not wrong, but not one i'd choose) but I guess you like perl, and more APIs don't hurt. It's a good example of adding more perl stuff, I guess.

File: src/fe-common/core/command-history.c (hunk 1 above)
Comment (ailin-nemui) [reply src/fe-common/core/command-history.c]: fixed

**Comment by ailin-nemui** [comment]
I renamed g_history to history_entries

**Pull request merged**